src/drivers/github.js

```diff
@@ -713,6 +722,72 @@ class Github {
   get userName() {
     return 'GitHub Action';
   }
+
+  async checkWorkflowForTimeoutMinutes() {
+    const WARNINGS = [];
+    const _DIR = '.github/workflows/';
+    const yaml = require('js-yaml');
+    const files = await fs.readdir(_DIR);
+    for (const file of files) {
+      const rawStr = await fs.readFile(`${_DIR}${file}`, 'utf8');
+      const doc = yaml.load(rawStr);
+      Object.keys(doc.jobs).forEach((job) => {
+        // does job contain timeout?
+        const timeoutVal = doc.jobs[job]['timeout-minutes'];
+        if (timeoutVal === undefined) return;
+        // does job have label runs-on "self-hosted" or "cml"
+        const runsOn = doc.jobs[job]['runs-on'];
+        switch (typeof runsOn) {
```

Comment:
not sure I follow the intention of this switch statement :thinking:

Reply:
it can a string or a list of strings